sale_order_lot_selection/view/sale_view.xml

```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="utf-8"?>
+<openerp>
+    <data>
+
+        <record id="view_order_tree_lot" model="ir.ui.view">
+            <field name="name">sale.order.tree.lot</field>
+            <field name="model">sale.order</field>
+            <field name="inherit_id" ref="sale.view_order_form"/>
+            <field name="arch" type="xml">
+                <xpath expr="//field[@name='order_line']/tree/field[@name='product_id']" position="after">
+                    <field name="lot_id" domain="[('product_id','=', product_id)]" context="{'product_id': product_id}"/>
+                </xpath>
+                <xpath expr="//field[@name='order_line']/form/group/group/field[@name='product_id']" position="after">
+                    <field name="lot_id" domain="[('product_id','=', product_id)]" context="{'product_id': product_id}"/>
```

Comment:
the field `lot_id` should only be visible to the members of the `stock.group_production_lot` res.groups